### DISCUSSION TITLE
feat: Calculate DependencyVersionDrift

### DISCUSSION THREAD

**Discussion by secustor** [renovatebot/renovate#26097] [Suggest an Idea]
I find myself often asking when onboarding new repos, how many versions actually a dependency currently is behind. 

1.2.3 -> 2.1.0
Could mean that 2 to x versions are between them. 

This number could be interesting to score repos by. 

Simply counting would be not perfect as if the dependency provides patches for multiple newer major versions. It is rather easy to implement though.

**Comment by HonkingGoose** [comment]
Yeah, it would be nice to get a quick feel on how badly you're behind. I expect Renovate users will get a shock when they update Renovate from `36.x` to latest `37.x` then: "I'm _how many_ versions behind?!?"

By the way, there's a typo in your final sentence: you meant `though` (as in but) but wrote `tough` (as in strong).

**Reply by secustor** [reply]
Fixed ;)

**Comment by rarkins** [comment]
Related: #17387

But please note as I commented there, we'd want to implement the algorithm ourselves cleanly and not based on the existing JS library which is GPL-3.0

**Reply by rarkins** [reply]
Paper which inspired libyear: https://ericbouwers.github.io/papers/icse15.pdf

**Comment by jamietanna** [comment]
@secustor I think this is now covered by LibYears, but note the caveats: https://github.com/renovatebot/renovate/issues/43303
